tools/packaging/static-build/shim-v2/build.sh

```diff
@@ -16,34 +16,45 @@ VMM_CONFIGS="qemu fc"
 
 GO_VERSION=${GO_VERSION}
 RUST_VERSION=${RUST_VERSION}
+CC=""
 
 DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
 container_image="${SHIM_V2_CONTAINER_BUILDER:-$(get_shim_v2_image_name)}"
 
+[ "${CROSS_BUILD}" == "true" ] && container_image_bk="${container_image}" && container_image="${container_image%x86_64}${ARCH}-cross-build"
 sudo docker pull ${container_image} || \
-       	(sudo docker build  \
+	(sudo docker ${BUILDX} build ${PLATFORM}  \
 		--build-arg GO_VERSION="${GO_VERSION}" \
 		--build-arg RUST_VERSION="${RUST_VERSION}" \
 		-t "${container_image}" \
 		"${script_dir}" && \
-	 push_to_registry "${container_image}")
+	push_to_registry "${container_image}")
 
-arch=$(uname -m)
+arch=${ARCH:-$(uname -m)}
 if [ ${arch} = "ppc64le" ]; then
 	arch="ppc64"
 fi
 
+#Build rust project using cross build musl image to speed up
+[[ "${CROSS_BUILD}" == "true" && ${ARCH} != "s390" ]] && container_image="messense/rust-musl-cross:${ARCH}-musl" && CC=${ARCH}-unknown-linux-musl-gcc
 sudo docker run --rm -i -v "${repo_root_dir}:${repo_root_dir}" \
+	--env CROSS_BUILD=${CROSS_BUILD} \
+	--env ARCH=${ARCH} \
+	--env CC="${CC}" \
 	-w "${repo_root_dir}/src/runtime-rs" \
 	"${container_image}" \
 	bash -c "git config --global --add safe.directory ${repo_root_dir} && make PREFIX=${PREFIX} QEMUCMD=qemu-system-${arch}"
 
 sudo docker run --rm -i -v "${repo_root_dir}:${repo_root_dir}" \
+	--env CROSS_BUILD=${CROSS_BUILD} \
+	--env ARCH=${ARCH} \
+	--env CC="${CC}" \
 	-w "${repo_root_dir}/src/runtime-rs" \
 	"${container_image}" \
 	bash -c "git config --global --add safe.directory ${repo_root_dir} && make PREFIX="${PREFIX}" DESTDIR="${DESTDIR}" install"
-	
+
+[ "${CROSS_BUILD}" == "true" ] && container_image="${container_image_bk}-cross-build"
```

Comment:
This step fails because the container_image value is considered as

> quay.io/kata-containers/builders:shim-v2-go-1.19.3-rust-1.66.0-41ed29bcd557362045831b870bdaab5109e5526c-dirty-x86_64-cross-build

instead of 

> quay.io/kata-containers/builders:shim-v2-go-1.19.3-rust-1.66.0-41ed29bcd557362045831b870bdaab5109e5526c-dirty-ppc64le-cross-build

It is because the value of `container_image_bk` which is fetched using `get_shim_v2_image_name` sets the ARCH via `uname -a`.
@jongwu, this needs to be set properly similar to setting `container_image="${container_image%x86_64}${ARCH}-cross-build"`